errors.go

```diff
@@ -2,147 +2,25 @@ package fuego
-	}
-	return e.Status
-}
-
-func (e HTTPError) DetailMsg() string {
-	return e.Detail
-}
-
-func (e HTTPError) Unwrap() error { return e.Err }
-
-// BadRequestError is an error used to return a 400 status code.
-type BadRequestError HTTPError
-
-var _ ErrorWithStatus = BadRequestError{}
-
-func (e BadRequestError) Error() string { return e.Err.Error() }
-
-func (e BadRequestError) StatusCode() int { return http.StatusBadRequest }
-
-func (e BadRequestError) Unwrap() error { return HTTPError(e) }
 
-// NotFoundError is an error used to return a 404 status code.
-type NotFoundError HTTPError
-
-var _ ErrorWithStatus = NotFoundError{}
-
-func (e NotFoundError) Error() string { return e.Err.Error() }
-
-func (e NotFoundError) StatusCode() int { return http.StatusNotFound }
-
-func (e NotFoundError) Unwrap() error { return HTTPError(e) }
-
-// UnauthorizedError is an error used to return a 401 status code.
-type UnauthorizedError HTTPError
-
-var _ ErrorWithStatus = UnauthorizedError{}
-
-func (e UnauthorizedError) Error() string { return e.Err.Error() }
-
-func (e UnauthorizedError) StatusCode() int { return http.StatusUnauthorized }
-
-func (e UnauthorizedError) Unwrap() error { return HTTPError(e) }
-
-// ForbiddenError is an error used to return a 403 status code.
-type ForbiddenError HTTPError
-
-var _ ErrorWithStatus = ForbiddenError{}
-
-func (e ForbiddenError) Error() string { return e.Err.Error() }
-
-func (e ForbiddenError) StatusCode() int { return http.StatusForbidden }
-
-func (e ForbiddenError) Unwrap() error { return HTTPError(e) }
-
-// ConflictError is an error used to return a 409 status code.
-type ConflictError HTTPError
-
-var _ ErrorWithStatus = ConflictError{}
-
-func (e ConflictError) Error() string { return e.Err.Error() }
-
-func (e ConflictError) StatusCode() int { return http.StatusConflict }
-
-func (e ConflictError) Unwrap() error { return HTTPError(e) }
-
-// InternalServerError is an error used to return a 500 status code.
-type InternalServerError = HTTPError
-
-// NotAcceptableError is an error used to return a 406 status code.
-type NotAcceptableError HTTPError
-
-var _ ErrorWithStatus = NotAcceptableError{}
-
-func (e NotAcceptableError) Error() string { return e.Err.Error() }
-
-func (e NotAcceptableError) StatusCode() int { return http.StatusNotAcceptable }
+	"github.com/go-fuego/fuego/internal"
+)
 
-func (e NotAcceptableError) Unwrap() error { return HTTPError(e) }
+type (
+	ErrorWithStatus     = internal.ErrorWithStatus
+	ErrorWithDetail     = internal.ErrorWithDetail
+	HTTPError           = internal.HTTPError
+	ErrorItem           = internal.ErrorItem
+	BadRequestError     = internal.BadRequestError
+	NotFoundError       = internal.NotFoundError
+	UnauthorizedError   = internal.UnauthorizedError
+	ForbiddenError      = internal.ForbiddenError
+	ConflictError       = internal.ConflictError
+	InternalServerError = internal.HTTPError
+	NotAcceptableError  = internal.NotAcceptableError
+)
```

Comment:
The commit/PR should mention the reason behind this change

https://github.com/go-fuego/fuego/issues/386#issuecomment-2677056026

https://github.com/go-fuego/fuego/issues/386#issuecomment-2677914205